include/nbl/video/IDescriptorPool.h

```diff
@@ -142,16 +159,19 @@ class IDescriptorPool : public core::IReferenceCounted, public IBackendObject
         friend class IGPUDescriptorSet;
         // Returns the offset into the pool's descriptor storage. These offsets will be combined
         // later with base memory addresses to get the actual memory address where we put the core::smart_refctd_ptr<const IDescriptor>.
-        SDescriptorOffsets allocateDescriptorOffsets(const IGPUDescriptorSetLayout* layout);
+        bool allocateDescriptorOffsets(SDescriptorOffsets& offsets, const IGPUDescriptorSetLayout* layout);
+        void freeDescriptorOffsets(SDescriptorOffsets& offsets, const IGPUDescriptorSetLayout* layout);
```

Comment:
why is this a free and not a cancel?

Reply:
Because it frees the descriptor offsets to cancel previously successful allocations. I don't know what "cancel descriptor offsets" would mean.

Reply:
yeah but its not an "arbitrary" free more like "abort most recent allocation" and name should reflect that.

Reply:
Resolved.